validator/main_test.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestExpandString(t *testing.T) {
```

Comment:
Can you include tests for all the examples from the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/file-configuration.md#environment-variable-substitution)?

Reply:
Also, would be good to smoke test this by validating the files in `/examples`. Once we're confident in this tooling can replace the root make `validate-examples` task with this tool.

Reply:
Added a `make validator-validate-examples`.

Reply:
I haven't added a test for each example yet. But I've added some, like `validator/shelltests/hex_integer.test`.

I will add the rest but wanted to check if you were ok with using `shelltest` and `yq` to verify. It just feels simpler to test e2e that the output result is correct by just running it from the command line than dealing with Go tests. But that is likely in large part my inexperience with Go testing :).

Reply:
I'm ok using shelltest and yq as long as its easy to run locally outside of just linux machines.